packages/plugins/src/api/index.ts

```diff
@@ -5,37 +5,47 @@
  */
 import { applyFilters, doAction } from '@wordpress/hooks';
 import { plugins as pluginsIcon } from '@wordpress/icons';
+import type { WPElement } from '@wordpress/element';
 
 /**
  * Defined behavior of a plugin type.
- *
- * @typedef {Object} WPPlugin
- *
- * @property {string}                    name    A string identifying the plugin. Must be
- *                                               unique across all registered plugins.
- * @property {string|WPElement|Function} [icon]  An icon to be shown in the UI. It can
- *                                               be a slug of the Dashicon, or an element
- *                                               (or function returning an element) if you
- *                                               choose to render your own SVG.
- * @property {Function}                  render  A component containing the UI elements
- *                                               to be rendered.
- * @property {string}                    [scope] The optional scope to be used when rendering inside
- *                                               a plugin area. No scope by default.
  */
+export interface WPPlugin {
+	/**
+	 * A string identifying the plugin. Must be unique across all registered plugins.
+	 */
+	name: string;
+
+	/**
+	 * An icon to be shown in the UI. It can be a slug of the Dashicon, or an
+	 * element (or function returning an element) if you choose to render your
+	 * own SVG.
+	 */
+	icon?: string | WPElement | Function;
```

Comment:
I would love to do this, but this is a great example of another rabbit hole. `IconType` isn't exported from wordpress components. So just looking into how to solve this type issue and make it perfect took another 20 minutes. 

It's valuable, I don't want to detract from that, but we probably don't have time to work on this project if we try to make every type perfect along the way. Mostly because I'm trying to cover a lot of ground relatively quickly

Reply:
The side problem is that it _shouldn't_ be hard to export a type from components, but there aren't any other types exported from there, so I don't have an example to follow for what's acceptable

Reply:
And another problem is that this introduces a dependency on wordpress components that isn't strictly required except for development, but we don't have any way to encode that since we don't allow dev dependencies in other packages